### PR TITLE
- Add new function `writeParameterStructureToXLS`

### LINKED ISSUE #670: New function `writeParametersToXLS`
`{esqlabsR}` has a function `readParametersFromXLS()` that reads parameter sets from an Excel file as supported by MoBi and returns a list containing vectors 'paths' with the full paths to the parameters, 'values' the values of the parameters, and 'units' with the units the values are in. This is the default "Parameter structure" used across the package.

There is currently no function that would do the opposite - write such a "parameter structure" into excel. The function `exportParametersToXLS` write `Parameter` objects, but not the list structure.

Proposed name for the new function `writeParameterStructureToXLS`? And then rename `readParametersFromXLS` to `readParameterStructureFromXLS` and deprecate `readParametersFromXLS` for a while?

@Felixmil @dlefaudeux @Laura-Villain what do you think? Need your input on usability.

### PR DESCRIPTION
**Pull request by PavelBal**
Fixes #670

It still lacks the implementation of appending to a file instead of overwriting, and maybe some tests. I would ask you to do this.

### CODE REVIEW

File: R/utilities-parameters.R [hunk 1]
```diff
@@ -39,6 +39,55 @@ readParametersFromXLS <- function(paramsXLSpath, sheets = NULL) {
...
 }
 
+#' Write parameter structure to excel that can be loaded in MoBi
+#'
+#' @param parameterStructure A list containing vectors 'paths' with the full paths to the parameters,
+#' 'values' the values of the parameters, and 'units' with the units the values are in.
+#'
+#' @param paramsXLSpath Path to the excel file
+#' @param sheet (Optional) name of the excel sheet
+#' @param extend If TRUE, the existing excel file will be extended with the new
```
Comment (PavelBal) [inline R/utilities-parameters.R:49]: Not implemented yet!

File: R/utilities-parameters.R [hunk 2]
```diff
@@ -39,6 +39,55 @@ readParametersFromXLS <- function(paramsXLSpath, sheets = NULL) {
...
+  )
+
+  # Write the results into an excel file.
+  # Wrap the output data frame into a list and name the list if sheet name
+  # has been provided
+  data <- list(output)
+  if (!is.null(sheet)) {
+    names(data) <- sheet
+  }
+  writeExcel(data = data, path = paramsXLSpath)
```
Comment (PavelBal) [inline R/utilities-parameters.R:88]: Make `writeExcel` internal? I think (and hope) that no one uses it directly.

Comment (Felixmil) [reply R/utilities-parameters.R:95]: done in a9e347a

File: R/utilities-parameters.R (hunk 1 above)
Comment (Felixmil) [reply R/utilities-parameters.R]: done in 964f738

**Review by Felixmil — APPROVED** [review]

File: NAMESPACE [hunk 3]
```diff
@@ -67,19 +67,15 @@ export(sourceAll)
 export(startFunctionVisualizer)
 export(startUnitConverter)
 export(stringToNum)
-export(writeExcel)
```
Comment (PavelBal) [inline NAMESPACE:70]: Add note to NEWS (breaking change?)

File: R/utilities-file.R [hunk 4]
```diff
@@ -66,7 +66,6 @@ readExcel <- function(path, sheet = NULL, ...) {
 #' @param data Data frame or named list of data frames that will be sheets in
 #' the xlsx
 #' @inheritParams writexl::write_xlsx
-#' @export
 writeExcel <- function(data, path, col_names = TRUE) {
```
Comment (PavelBal) [inline R/utilities-file.R:69]: prepend `.` to the name to stay consistent with naming conventions.

Comment (Felixmil) [reply R/utilities-file.R]: done in a2f2994

File: NAMESPACE (hunk 3 above)
Comment (Felixmil) [reply NAMESPACE:70]: done in a2f2994

**Pull request merged**